go/vt/vtgate/buffer/buffer_test.go

```diff
@@ -782,3 +795,66 @@ func testShutdown1(t *testing.T, fail failover) {
 		t.Fatal(err)
 	}
 }
+
+func TestParallelRangeIndex(t *testing.T) {
+	suite := []struct {
+		max         int
+		concurrency int
+		calls       []int
+	}{
+		{
+			max:         0,
+			concurrency: 0,
+			calls:       []int{},
+		},
+		{
+			max:         100,
+			concurrency: 0,
+			calls:       []int{},
+		},
+		{
+			max:         9,
+			concurrency: 3,
+			calls:       []int{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		},
+		{
+			max:         0,
+			concurrency: 10,
+			calls:       []int{0},
+		},
+		{
+			max:         9,
+			concurrency: 9,
+			calls:       []int{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		},
+	}
+
+	for idx, tc := range suite {
+		name := fmt.Sprintf("%d_max%d_concurrency%d", idx, tc.max, tc.concurrency)
+		t.Run(name, func(t *testing.T) {
+			var mu sync.Mutex
+			var wg sync.WaitGroup
+			var counter atomic.Int64
+
+			wg.Add(tc.concurrency)
+			got := []int{}
```

Comment:
Annoyingly nitty, but we can pre-allocate the space one time:
```
			got := make([]int, 0, len(tc.calls))
```

Reply:
Yeah our style guide recommends either an empty variable (`var got []int`) or explicitly allocating the slice. Either way is fine.